File: 1256-rank-transform-of-an-array/rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        map = {}
        copy_arr = sorted(arr)
        rank = 1
        ans = []

        for i in range(0, len(copy_arr)):
            if i > 0 and copy_arr[i] > copy_arr[i - 1]:
                rank = rank + 1
            map[copy_arr[i]] = rank
        for i in range(0, len(arr)):
            ans.append(map[arr[i]])
        return ans
        

File: 1256-rank-transform-of-an-array/rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> copy_arr = arr;
        std::unordered_map<int, int> ump;
        vector<int> ans;

        if (arr.size() == 0) return ans;
        sort(copy_arr.begin(),copy_arr.end());
        ump[copy_arr[0]] = 1;
        for(int i = 1, rank = 2; i < copy_arr.size(); i++) {
            if (copy_arr[i] > copy_arr[i - 1]) {
                ump[copy_arr[i]] = rank;
                rank++; 
            }
        }
        for(int i = 0; i < arr.size(); i++) {
            ans.push_back(ump[arr[i]]);
        }
        return ans;
    }
};